### PR TITLE
Logfmt parser

### LINKED ISSUE #4427: Add logfmt Parser Plugin
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:
add a parser for `logfmt`

### Current behavior:
None

### Desired behavior:
the ability to parse logfmt log messages and specify the keys for timestamp (and format), tags, and fields

### Use case: [Why is this important (helps with prioritizing requests)]
this would be useful to ingesting influxdb log data

### PR DESCRIPTION
**Pull request by Ayrdrie**
### Required for all PRs:
 closes #4427 

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [X] Has appropriate unit tests.

### CODE REVIEW

File: docs/DATA_FORMATS_INPUT.md [hunk 1]
```diff
@@ -886,4 +887,25 @@ the file output will only print once per `flush_interval`.
 - If successful, add the next token, update the pattern and retest.
 - Continue one token at a time until the entire line is successfully parsed.
 
+# Logfmt
+This parser implements the logfmt format by extracting key-value pairs from log text in the form `<key>=<value>`.
+At the moment, the plugin will produce one metric per line and all keys
+are added as fields. Values are left as strings (for now).
```
Comment (danielnelson) [inline docs/DATA_FORMATS_INPUT.md]: > Values are left as strings (for now).

This isn't true, we are doing auto conversion to other types.

File: plugins/parsers/logfmt/parser.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+type Parser struct {
+	MetricName  string
+	DefaultTags map[string]string
+	Now         func() time.Time
+}
+
+// NewParser creates a parser.
+func NewParser(metricName string, defaultTags map[string]string) *Parser {
+	return &Parser{
+		MetricName:  metricName,
```
Comment (danielnelson) [inline plugins/parsers/logfmt/parser.go]: What happens if this is not set?

File: docs/DATA_FORMATS_INPUT.md [hunk 3]
```diff
@@ -886,4 +887,25 @@ the file output will only print once per `flush_interval`.
...
+  data_format = "logfmt"
+  ```
 
+A typical log
+
+method=GET host=influxdata.org ts=2018-07-24T19:43:40.275Z connect=4ms service=8ms status=200 bytes=1653
+```
+will be converted into
+```
+method=GET, host=influxdata.org, ts=2018-07-24T19:43:40.275Z, connect=4ms, service=8ms, status=200, bytes=1653,
```
Comment (danielnelson) [inline docs/DATA_FORMATS_INPUT.md]: Turn this into valid line protocol:
```
logfmt method="GET",host="influxdata.org",ts="2018-07-24T19:43:40.275Z",connect="4ms",service="8ms",status=200i,bytes=1653i
```

File: plugins/parsers/logfmt/parser.go (hunk 2 above)
Comment (maxunt) [reply plugins/parsers/logfmt/parser.go]: By default, doesn't telegraf use the plugin name of the input/processor that uses it?

**Pull request merged**